core/src/test/java/com/graphhopper/routing/HeadingRoutingTest.java

```diff
@@ -164,6 +166,52 @@ public void headingTest5() {
         assertArrayEquals(new int[]{5, 4, 3, 8, 7, 6, 5, 4, 3, 2}, calcNodes(graph, response.getAll().get(0)));
     }
 
+    @Test
+    public void headingTestWithToleranceRoundTrip() {
+        // Test enforce start direction
+        GraphHopperStorage graph = createSquareGraph();
+        int tolerance = 80;
+        Router router = createRouter(graph);
```

Comment:
In this test I would calculate two routes with two different values for the tolerance. This way we can be sure the tolerance actually makes a difference (now and in the future when some of the code changes).